server/validators/dateIsSameOrAfterOtherProperty.ts

```diff
@@ -0,0 +1,21 @@
+import { registerDecorator, ValidationArguments, ValidationOptions } from 'class-validator'
+import SimpleDate, { simpleDateFromDate } from '../commonValidationTypes/simpleDate'
+
+export default function DateIsSameOrAfterOtherProperty(property: string, validationOptions?: ValidationOptions) {
+  return (object: unknown, propertyName: string) => {
+    registerDecorator({
+      name: 'dateIsSameOrAfterOtherProperty',
+      target: object.constructor,
+      propertyName,
+      constraints: [property],
+      options: validationOptions,
+      validator: {
+        validate(value: SimpleDate, args: ValidationArguments) {
+          const [relatedPropertyName] = args.constraints
+          const relatedValue = args.object[relatedPropertyName]
+          return relatedValue ? value >= simpleDateFromDate(new Date(relatedValue)) : true
```

Comment:
I'm guessing this works because JavaScript is automatically invoking SimpleDate's `.toString()` method to run this comparison, and the `.toString()` method is returning the date in a format which can be assessed with the greater than operator.

The implicit string conversion top of the use of a greater than operator on a string value makes me feel a bit uneasy. No need to update though.. I think this should work and is probably the most concise way to write this logic honestly. I just think I would have preferred to compare on the date object itself with, https://date-fns.org/v2.30.0/docs/isAfter